packages/core/elasticsearch/core-elasticsearch-server-internal/src/version_check/ensure_es_version.ts

```diff
@@ -164,9 +180,15 @@ export const pollEsNodesVersion = ({
         })
       );
     }),
-    map((nodesInfoResponse: NodesInfo & { nodesInfoRequestError?: Error }) =>
-      mapNodesVersionCompatibility(nodesInfoResponse, kibanaVersion, ignoreVersionMismatch)
-    ),
-    distinctUntilChanged(compareNodes) // Only emit if there are new nodes or versions or if we return an error and that error changes
+    map((nodesInfoResponse: NodesInfo & { nodesInfoRequestError?: Error }) => {
+      return mapNodesVersionCompatibility(nodesInfoResponse, kibanaVersion, ignoreVersionMismatch);
+    }),
+    // Only emit if there are new nodes or versions or if we return an error and that error changes
+    distinctUntilChanged(compareNodes),
+    tap((nodesVersionCompatibility) => {
+      if (nodesVersionCompatibility.isCompatible) {
```

Comment:
I don't know if it could happen in real life, but if for some reason ES nodes become _incompatible_, we could go back to faster "startup" checking again:
```typescript
tap((nodesVersionCompatibility) => isStartup$.next(!nodesVersionCompatibility.isCompatible))
```